ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -0,0 +1,288 @@
+        auto const& sigma_eff = _ip_data[ip].sigma_eff;
+
+        auto const alpha = _process_data.biot_coefficient(t, x_position)[0];
+        auto const rho_sr = _process_data.solid_density(t, x_position)[0];
+        auto const rho_fr = _process_data.fluid_density(t, x_position)[0];
+        auto const porosity = _process_data.porosity(t, x_position)[0];
+        auto const& b = _process_data.specific_body_force;
+        auto const& identity2 = MaterialLib::SolidModels::Invariants<
+            KelvinVectorDimensions<DisplacementDim>::value>::identity2;
+
+        eps.noalias() = B * u;
+
+        auto C = _ip_data[ip].updateConstitutiveRelation(t, x_position, dt, u);
+
+        local_Jac.noalias() += B.transpose() * C * B * w;
+
+        double p_at_xi = 0.;
+        NumLib::shapeFunctionInterpolate(local_p, N_p, p_at_xi);
+
+        double const rho = rho_sr * (1 - porosity) + porosity * rho_fr;
+        local_rhs.noalias() -=
+            (B.transpose() * (sigma_eff - alpha * identity2 * p_at_xi) -
+             N_u_op.transpose() * rho * b) *
+            w;
+    }
+}
+
+template <typename ShapeFunctionDisplacement, typename ShapeFunctionPressure,
+          typename IntegrationMethod, int DisplacementDim>
+void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
+                                  ShapeFunctionPressure, IntegrationMethod,
+                                  DisplacementDim>::
+    assembleWithJacobianAndCoupling(
+        const double t,
+        const std::vector<double>& local_xdot,
+        const double dxdot_dx,
+        const double dx_dx,
+        std::vector<double>& local_M_data,
+        std::vector<double>& local_K_data,
+        std::vector<double>& local_b_data,
+        std::vector<double>& local_Jac_data,
+        const LocalCoupledSolutions& local_coupled_solutions)
+{
+    // For the equations with pressure
+    if (local_coupled_solutions.process_id == 0)
+    {
+        assembleWithJacobianForPressureEquations(
+            t, local_xdot, dxdot_dx, dx_dx, local_M_data, local_K_data,
+            local_b_data, local_Jac_data, local_coupled_solutions);
+        return;
+    }
+
+    // For the equations with deformation
+    assembleWithJacobianForDeformationEquations(
+        t, local_xdot, dxdot_dx, dx_dx, local_M_data, local_K_data,
+        local_b_data, local_Jac_data, local_coupled_solutions);
+}
+
+template <typename ShapeFunctionDisplacement, typename ShapeFunctionPressure,
+          typename IntegrationMethod, int DisplacementDim>
+void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
+                                  ShapeFunctionPressure, IntegrationMethod,
+                                  DisplacementDim>::
+    postNonLinearSolverConcrete(std::vector<double> const& local_x,
+                                double const t,
+                                bool const use_monolithic_scheme)
+{
+    const int offset = use_monolithic_scheme ? displacement_index : 0;
```

Comment:
better to call it displacement_offset to be consistent with displacement_size and displacement_index.

Reply:
Changed accordingly.